e2e/ctl_v3_member_test.go

```diff
@@ -101,3 +103,66 @@ func ctlV3MemberRemove(cx ctlCtx, memberID, clusterID string) error {
 	cmdArgs := append(cx.PrefixArgs(), "member", "remove", memberID)
 	return spawnWithExpect(cmdArgs, fmt.Sprintf("%s removed from cluster %s", memberID, clusterID))
 }
+
+func memberAddTest(cx ctlCtx) {
+	peerURL := fmt.Sprintf("http://localhost:%d", etcdProcessBasePort+11)
+	cmdArgs := append(cx.PrefixArgs(), "member", "add", "newmember", fmt.Sprintf("--peerURLs=%s", peerURL))
+	if err := spawnWithExpect(cmdArgs, " added to cluster "); err != nil {
+		cx.t.Fatal(err)
+	}
+
+	mresp, err := getMemberList(cx)
+	if err != nil {
+		cx.t.Fatal(err)
+	}
+	if len(mresp.Members) != 2 {
+		cx.t.Fatalf("expected 2, got %d", len(mresp.Members))
+	}
+
+	found := false
+	for _, mem := range mresp.Members {
+		for _, v := range mem.PeerURLs {
+			if v == peerURL {
+				found = true
+				break
+			}
+		}
+	}
+	if !found {
+		cx.t.Fatalf("expected %s in PeerURLs, got %+v", peerURL, mresp.Members)
+	}
+}
+
+func memberUpdateTest(cx ctlCtx) {
+	mr, err := getMemberList(cx)
+	if err != nil {
+		cx.t.Fatal(err)
+	}
+
+	peerURL := fmt.Sprintf("http://localhost:%d", etcdProcessBasePort+11)
+	cmdArgs := append(cx.PrefixArgs(), "member", "update", fmt.Sprintf("%x", mr.Members[0].ID), fmt.Sprintf("--peerURLs=%s", peerURL))
+	if err = spawnWithExpect(cmdArgs, " updated in cluster "); err != nil {
+		cx.t.Fatal(err)
+	}
+
+	mresp, err := getMemberList(cx)
+	if err != nil {
+		cx.t.Fatal(err)
+	}
+	if len(mresp.Members) != 1 {
+		cx.t.Fatalf("expected 1, got %d", len(mresp.Members))
+	}
+
+	found := false
+	for _, mem := range mresp.Members {
```

Comment:
since it's only one member I think this loop can be replaced with something like `if mresp.Members[0].PeerURLs[0] != newPeerURL { fatal }`